forge-game/src/main/java/forge/game/card/Card.java

```diff
@@ -4316,6 +4319,7 @@ public final void addChangedCardTypesByText(final CardType addType, final long t
         this.changedTextColors.addEmpty(timestamp, staticId);
         this.changedTextTypes.addEmpty(timestamp, staticId);
 
+        this.updateTypeCache();
```

Comment:
maybe move to below function instead
you're missing at least `addChangedTextTypeWord`